lib/THNN/generic/SpatialAveragePooling.c

```diff
@@ -74,7 +62,7 @@ static int nn_(SpatialAveragePooling_updateOutput)(lua_State *L)
     THTensor_(resize4d)(output, input->size[0], nInputPlane, outputHeight, outputWidth);
   
   input = THTensor_(newContiguous)(input);
-  luaL_argcheck(L, THTensor_(isContiguous)(output), 1, "");
+  THArgCheck(THTensor_(isContiguous)(output), 1, "");
```

Comment:
This was already the case in the old impl? Do you have an idea why no error message is provide and argument index is 1 (e.g. output was a field before but now is a real arg)? Same goes for ln. 177.